include/core/Attribute.hpp

```diff
@@ -9,24 +8,44 @@ namespace UML {
 
 enum ACCESS { PUBLIC, PRIVATE, PROTECTED };
 
+class AccessType {
+  ACCESS m_access;
+  AccessType(ACCESS aAccess) : m_access(aAccess) {}
+
+ public:
+  ~AccessType() = default;
+  ACCESS getType() const { return m_access; }
+  std::string toString() const;
+  static AccessType *getInstance(ACCESS aAccess);
+  friend std::ostream &operator<<(std::ostream &os, const AccessType &obj);
+};
+
+namespace AccessTypeProvider {
+extern AccessType *m_public;
+extern AccessType *m_protected;
+extern AccessType *m_private;
```

Comment:
Add a function maybe 'runCleanUpTasks' which will free this memory. Or make use of smart pointers.